tidy3d/plugins/smatrix/smatrix.py

```diff
@@ -36,7 +36,7 @@ class Port(Box):
         title="Mode Specification",
         description="Specifies how the mode solver will solve for the modes of the port.",
     )
-    mode_indices: Tuple[pd.NonNegativeInt, ...] = pd.Field(
+    mode_indices: Optional[Tuple[pd.NonNegativeInt, ...]] = pd.Field(
```

Comment:
I wonder if we should remove this for simplicity? We just run all modes in the ModeSpec. If a user wants to only *source* a subset of the modes, we now have the `run_only` parameter. If they want to use modes that are not in order, e.g. 0 and 2, they can just add two ports.

One thing that is a bit confusing at first and made me think about this is that I was wondering if `MatrixIndex` indexes into the ModeSpec-defined array of indexes (`np.arange(mode_spec.num_modes)`) or the `Port.mode_indeces` subset of that array. I assume the latter.

Reply:
agreed, removed this.